Autopilot/Src/xbee.cpp

```diff
@@ -0,0 +1,43 @@
+#include "xbee.hpp"
+#include "stm32f7xx_hal_usart.h"
+
+extern USART_HandleTypeDef husart2;
+
+XBEE* XBEE::xbeeInstance = nullptr;
+
+XBEE* XBEE::getInstance() {
+    if (!xbeeInstance) {
+        xbeeInstance = new XBEE();
+    }
+
+    return xbeeInstance;
+}
+
+XBEE::XBEE() {}
+
+void XBEE::Receive_GS_Data() {}
+
+void XBEE::Send_GS_Data() {}
+
+xbee_in_t GetResult() {
+	uint8_t holder;
+	xbee_in_t xbeeIn;
+	HAL_USART_Receive(&husart2, &holder, 1, 100);
+	HAL_USART_Receive(&husart2, &xbeeIn.len, 1, 100);
```

Comment:
1. Calling to the xbees should be done in `Receive_GS_Data()`. This method is a getter, so its role is to just send information out, not gather it
2. `HAL_USART_Receive` is blocking. refer to other comment. 

I don't think this will behave as you expect too... 

If this is anything like GPS, then the xbees will be constantly spitting data at you. This means at any given time you cannot be sure if you are getting the length or some other data. I feel like using DMA and a parser like I did with GPS would be your best bet. We can call to discuss more

@Fberbari you have any experience or say here?